app/controllers/rooms_controller.rb

```diff
@@ -40,14 +40,27 @@ def check_available
   private
 
   def filtered_rooms
-    Room.includes(:facilities)
-        .filter_by_room_type(params[:room_type])
-        .filter_by_view_type(params[:view_type])
-        .available_in_priod(params[:check_in], params[:check_out])
-        .send(params[:sort].presence || "latest")
+    RoomType.availabel_rooms(params[:check_in], params[:check_out])
+            .send(params[:sort].presence || "latest")
```

Comment:
Quên mất. Tk send này nó access được vào private method nên để an toàn thì dùng public_send nhé